### PR TITLE
e2e: verify generated pprof files

### PR DESCRIPTION
**Pull request by alebedev87**
Enhance the e2e test with the actual validation of NodeObservability Operator's outcome: pprof files.

### CODE REVIEW

**Comment by alebedev87** [comment]
GCP cluster install failed

/retest

**Comment by alebedev87** [comment]
/label qe-approved
/label docs-approved
/label px-approved

Just an improvement to existing e2e test cases. No impact on the operator's code or docs.

File: pkg/operator/controller/nodeobservability/daemonset_util.go [hunk 1]
```diff
@@ -11,6 +11,12 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
+// LabelsForNodeObservability returns the labels for selecting the resources
+// belonging to the given node observability CR name.
+func LabelsForNodeObservability(name string) map[string]string {
```
Comment (thejasn) [inline pkg/operator/controller/nodeobservability/daemonset_util.go:16]: nit: I would still prefer this to be private and explicitly added in the test just to decouple them.

Comment (alebedev87) [reply pkg/operator/controller/nodeobservability/daemonset_util.go]: Done

**Pull request merged**